src/p2p/net_node.inl

```diff
@@ -1095,19 +1161,22 @@ namespace nodetool
     {
       if(conn_count < expected_white_connections)
       {
-        //start from white list
-        if(!make_expected_connections_count(true, expected_white_connections))
+        //start from anchor list
+        if(!make_expected_connections_count(anchor, P2P_DEFAULT_ANCHOR_CONNECTIONS_COUNT))
           return false;
-        //and then do grey list
-        if(!make_expected_connections_count(false, m_config.m_net_config.connections_count))
+        //then do white list
+        if(!make_expected_connections_count(white, expected_white_connections))
```

Comment:
I did not want to change too much the current behaviour and anchors are a subset of white peers, so I did it that way.

Master: ```Anchor 0 White 5 Gray 3```
PR: ```Anchor 2 White 3 Gray 3```

I can change it to: ```Anchor 2 White 5 Gray 3``` and increase ```P2P_DEFAULT_CONNECTIONS_COUNT``` to 10.